tests/s2n_test.h

```diff
@@ -211,6 +211,15 @@ int test_count;
 #define EXPECT_STRING_EQUAL( p1, p2 ) EXPECT_EQUAL( strcmp( (p1), (p2) ), 0 )
 #define EXPECT_STRING_NOT_EQUAL( p1, p2 ) EXPECT_NOT_EQUAL( strcmp( (p1), (p2) ), 0 )
 
+#define EXPECT_NOT_BLOCKED(conn, blocked, expected_msg) \
+    EXPECT_EQUAL((blocked), S2N_NOT_BLOCKED);           \
+    EXPECT_EQUAL(s2n_conn_get_current_message_type(conn), (expected_msg))
+#define EXPECT_BLOCKED_ON_EARLY_DATA(result) EXPECT_FAILURE_WITH_ERRNO((result), S2N_ERR_EARLY_DATA_BLOCKED)
+#define EXPECT_BLOCKED_ON_IO(result)         EXPECT_FAILURE_WITH_ERRNO((result), S2N_ERR_IO_BLOCKED)
+#define EXPECT_BLOCKED_ON(conn, blocked, expected_blocked, expected_msg) \
+    EXPECT_EQUAL((blocked), (expected_blocked));                         \
+    EXPECT_EQUAL(s2n_conn_get_current_message_type(conn), (expected_msg))
```

Comment:
These definitely don't belong in s2n_test.h. Minimally, move them to s2n_testlib.h. s2n_test.h should only be very minimal common asserts.

I'd also worry these were too specific to the file you got them from. Certainly EXPECT_BLOCKED_ON_EARLY_DATA isn't going to be widely used :) EXPECT_BLOCKED_ON_IO maybe, but that's not much simpler than just doing the EXPECT_FAILURE_WITH_ERRNO yourself. And all the names probably need another look if these are going to be common methods. Minimally the ones that expect a specific handshake message should mention they're for use during the handshake / when calling s2n_negotiate.